hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngrade.java

```diff
@@ -95,6 +95,16 @@ public static boolean needsUpgrade(HoodieTableMetaClient metaClient, HoodieWrite
       throw new HoodieUpgradeDowngradeException(
           String.format("Please upgrade table from version %s to %s before upgrading to version %s.", fromTableVersion, HoodieTableVersion.SIX.versionCode(), toWriteVersion));
     }
+    // for table versions greater than or equal to six, if we are attempting to do an upgrade and auto-upgrade is disabled
```

Comment:
I think it's better for `needsUpgrade` to take only one of `HoodieWriteConfig config` and `HoodieTableVersion toWriteVersion` for validation because `config` can contain write table version, which can be confusing now.